content/src/service/synchronization/ContentCluster.ts

```diff
@@ -175,22 +186,50 @@ export class ContentCluster {
     }
 
     /** Detect my own identity */
-    private async detectMyIdentity(servers: Set<ServerMetadata>): Promise<void> {
+    async detectMyIdentity(attempts: number = 1): Promise<void> {
         try {
-            // Ask each server for their name
-            const serverNames = await Promise.all(Array.from(servers)
-                .map(async serverMetadata => ({ metadata: serverMetadata, name: await this.getServerName(serverMetadata.address) })))
-
-            // Filter out other servers
-            const serversWithMyName = serverNames.filter(({ name }) => name == this.nameKeeper.getServerName())
-
-            if (serversWithMyName.length > 1) {
-                ContentCluster.LOGGER.warn(`Expected to find only one server with my name '${this.nameKeeper.getServerName()}', but found ${serversWithMyName.length}`)
-            } else {
-                this.myIdentity = serversWithMyName[0]?.metadata
+            // Fetch server list from the DAO
+            const allServers: Set<ServerMetadata> = await this.dao.getAllContentServers()
+            const serversByAddresses: Map<ServerAddress, ServerMetadata> = new Map(Array.from(allServers).map(metadata => [metadata.address, metadata]))
+            const challengesByAddress: Map<ServerAddress, ChallengeText> = new Map()
```

Comment:
I think this can be class cache. Challenges should not change, right?

Reply:
Yes, they change when a server is restarted. Since it's only an HTTP call, I think we can make it again, right? Or are you worried about something in particular?